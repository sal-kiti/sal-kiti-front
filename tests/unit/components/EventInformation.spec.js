import { createLocalVue, shallowMount } from "@vue/test-utils";
import flushPromises from "flush-promises";
import EventInformation from "@/components/EventInformation.vue";
import BootstrapVue from "bootstrap-vue";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(BootstrapVue);
localVue.use(Vuex);

let state = { user: { is_authenticated: false, manager: [] } };
let store = new Vuex.Store({ state });

describe("EventInformation.vue", () => {
  it("renders event information", async () => {
    const $route = {
      params: { event_id: 1 }
    };
    const wrapper = shallowMount(EventInformation, {
      localVue,
      store,
      mocks: {
        $route
      }
    });
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  });

  it("show management options for staff", async () => {
    state = {
      editMode: true,
      user: { is_authenticated: true, is_staff: true, manager: [] }
    };
    store = new Vuex.Store({ state });
    const $route = {
      params: { event_id: 1 }
    };
    const wrapper = shallowMount(EventInformation, {
      localVue,
      store,
      mocks: {
        $route
      }
    });
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  });

  it("show management options for organization user", async () => {
    state = { editMode: true, user: { is_authenticated: true, manager: [] } };
    store = new Vuex.Store({ state });
    const $route = {
      params: { event_id: 1 }
    };
    const wrapper = shallowMount(EventInformation, {
      localVue,
      store,
      mocks: {
        $route
      }
    });
    await flushPromises();
    wrapper.vm.$data.event.permissions = { update: true };
    expect(wrapper.element).toMatchSnapshot();
  });

  it("swap visibility", async () => {
    state = { user: { is_authenticated: true, manager: [] } };
    store = new Vuex.Store({ state });
    const $route = {
      params: { event_id: 1 }
    };
    const wrapper = shallowMount(EventInformation, {
      localVue,
      store,
      mocks: {
        $route
      }
    });
    await flushPromises();
    expect(wrapper.vm.$data.event.public).toBe(false);
    await wrapper.vm.togglePublicStatus();
    expect(wrapper.vm.$data.event.public).toBe(true);
  });

  it("swap lock", async () => {
    state = { user: { is_authenticated: true, manager: [] } };
    store = new Vuex.Store({ state });
    const $route = {
      params: { event_id: 1 }
    };
    const wrapper = shallowMount(EventInformation, {
      localVue,
      store,
      mocks: {
        $route
      }
    });
    await flushPromises();
    expect(wrapper.vm.$data.event.locked).toBe(false);
    await wrapper.vm.toggleLockStatus();
    expect(wrapper.vm.$data.event.locked).toBe(true);
  });

  it("render error message on error", async () => {
    const $route = {
      params: { event_id: 99 }
    };
    const wrapper = shallowMount(EventInformation, {
      localVue,
      mocks: {
        $route
      },
      store
    });
    await flushPromises();
    expect(wrapper.text()).toContain("import.error.unknown");
  });
});
