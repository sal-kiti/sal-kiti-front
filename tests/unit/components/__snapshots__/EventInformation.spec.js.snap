// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventInformation.vue renders event information 1`] = `
<div>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h1>
        event.event: Demo Champion
      </h1>
    </b-col-stub>
     
    <!---->
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h2
        class="bg-sal-orange"
      >
        event.info
      </h2>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <!---->
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      cols="12"
      tag="div"
    >
      <dl>
        <dt>
          description
        </dt>
         
        <dd>
          Demo competition on the Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <!---->
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          date
        </dt>
         
        <dd>
          2019-01-22 - 2019-01-24
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          organizer
        </dt>
         
        <dd>
          
          Local Club
        
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          location
        </dt>
         
        <dd>
          Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
  </b-row-stub>
   
  <!---->
</div>
`;

exports[`EventInformation.vue show management options for area managers 1`] = `
<div>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h1>
        event.event: Demo Champion
      </h1>
    </b-col-stub>
     
    <b-col-stub
      tag="div"
    >
      <b-button-stub
        class="btn-orange float-right"
        tag="button"
        target="_self"
        to="[object Object]"
        type="button"
        variant="light"
      >
        event.edit
      
      </b-button-stub>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h2
        class="bg-sal-orange"
      >
        event.info
      </h2>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <!---->
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      cols="12"
      tag="div"
    >
      <dl>
        <dt>
          description
        </dt>
         
        <dd>
          Demo competition on the Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <!---->
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          date
        </dt>
         
        <dd>
          2019-01-22 - 2019-01-24
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          organizer
        </dt>
         
        <dd>
          
          Local Club
        
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          location
        </dt>
         
        <dd>
          Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="3"
      tag="div"
    >
      <dl>
        <dt>
          event.safety_plan
        </dt>
         
        <dd>
          
          no
        
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <b-col-stub
      cols="12"
      tag="div"
    >
      <dl>
        <dt>
          event.notes
        </dt>
         
        <dd>
          
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          competition.status
        </dt>
         
        <!---->
         
        <dd>
          
          competition.unlocked
          
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.lock
          
          </b-button-stub>
           
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.lock
            (event.include_competitions)
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          event.approved
        </dt>
         
        <!---->
         
        <dd>
          
          no
          
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.approve
          
          </b-button-stub>
           
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.approve
            (event.include_competitions)
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          competition.visibility
        </dt>
         
        <!---->
         
        <dd>
          
          competition.hidden
          
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.publish
          
          </b-button-stub>
           
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.publish
            (event.include_competitions)
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
  </b-row-stub>
   
  <!---->
</div>
`;

exports[`EventInformation.vue show management options for organization user 1`] = `
<div>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h1>
        event.event: Demo Champion
      </h1>
    </b-col-stub>
     
    <b-col-stub
      tag="div"
    >
      <b-button-stub
        class="btn-orange float-right"
        tag="button"
        target="_self"
        to="[object Object]"
        type="button"
        variant="light"
      >
        event.edit
      
      </b-button-stub>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h2
        class="bg-sal-orange"
      >
        event.info
      </h2>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <!---->
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      cols="12"
      tag="div"
    >
      <dl>
        <dt>
          description
        </dt>
         
        <dd>
          Demo competition on the Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <!---->
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          date
        </dt>
         
        <dd>
          2019-01-22 - 2019-01-24
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          organizer
        </dt>
         
        <dd>
          
          Local Club
        
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          location
        </dt>
         
        <dd>
          Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <eventformcontacts-stub
        eventid="1"
      />
    </b-col-stub>
  </b-row-stub>
</div>
`;

exports[`EventInformation.vue show management options for sport managers 1`] = `
<div>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h1>
        event.event: Demo Champion
      </h1>
    </b-col-stub>
     
    <b-col-stub
      tag="div"
    >
      <b-button-stub
        class="btn-orange float-right"
        tag="button"
        target="_self"
        to="[object Object]"
        type="button"
        variant="light"
      >
        event.edit
      
      </b-button-stub>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h2
        class="bg-sal-orange"
      >
        event.info
      </h2>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <!---->
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      cols="12"
      tag="div"
    >
      <dl>
        <dt>
          description
        </dt>
         
        <dd>
          Demo competition on the Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <!---->
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          date
        </dt>
         
        <dd>
          2019-01-22 - 2019-01-24
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          organizer
        </dt>
         
        <dd>
          
          Local Club
        
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          location
        </dt>
         
        <dd>
          Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="3"
      tag="div"
    >
      <dl>
        <dt>
          event.safety_plan
        </dt>
         
        <dd>
          
          no
        
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <b-col-stub
      cols="12"
      tag="div"
    >
      <dl>
        <dt>
          event.notes
        </dt>
         
        <dd>
          
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          competition.status
        </dt>
         
        <!---->
         
        <dd>
          
          competition.unlocked
          
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.lock
          
          </b-button-stub>
           
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.lock
            (event.include_competitions)
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          event.approved
        </dt>
         
        <!---->
         
        <dd>
          
          no
          
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.approve
          
          </b-button-stub>
           
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.approve
            (event.include_competitions)
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          competition.visibility
        </dt>
         
        <!---->
         
        <dd>
          
          competition.hidden
          
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.publish
          
          </b-button-stub>
           
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.publish
            (event.include_competitions)
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
  </b-row-stub>
   
  <!---->
</div>
`;

exports[`EventInformation.vue show management options for staff 1`] = `
<div>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h1>
        event.event: Demo Champion
      </h1>
    </b-col-stub>
     
    <b-col-stub
      tag="div"
    >
      <b-button-stub
        class="btn-orange float-right"
        tag="button"
        target="_self"
        to="[object Object]"
        type="button"
        variant="light"
      >
        event.edit
      
      </b-button-stub>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h2
        class="bg-sal-orange"
      >
        event.info
      </h2>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <!---->
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      cols="12"
      tag="div"
    >
      <dl>
        <dt>
          description
        </dt>
         
        <dd>
          Demo competition on the Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <!---->
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          date
        </dt>
         
        <dd>
          2019-01-22 - 2019-01-24
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          organizer
        </dt>
         
        <dd>
          
          Local Club
        
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          location
        </dt>
         
        <dd>
          Village Field
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="3"
      tag="div"
    >
      <dl>
        <dt>
          event.safety_plan
        </dt>
         
        <dd>
          
          no
        
        </dd>
      </dl>
    </b-col-stub>
     
    <!---->
     
    <b-col-stub
      cols="12"
      tag="div"
    >
      <dl>
        <dt>
          event.notes
        </dt>
         
        <dd>
          
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          competition.status
        </dt>
         
        <!---->
         
        <dd>
          
          competition.unlocked
          
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.lock
          
          </b-button-stub>
           
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.lock
            (event.include_competitions)
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          event.approved
        </dt>
         
        <!---->
         
        <dd>
          
          no
          
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.approve
          
          </b-button-stub>
           
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.approve
            (event.include_competitions)
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          competition.visibility
        </dt>
         
        <!---->
         
        <dd>
          
          competition.hidden
          
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.publish
          
          </b-button-stub>
           
          <b-button-stub
            class="m-1"
            size="sm"
            tag="button"
            target="_self"
            type="button"
            variant="outline-success"
          >
            
            competition.publish
            (event.include_competitions)
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
     
    <b-col-stub
      cols="6"
      md="3"
      tag="div"
    >
      <dl>
        <dt>
          admin
        </dt>
         
        <dd>
          <b-button-stub
            class="btn-orange btn-sm"
            tag="button"
            target="_self"
            to="[object Object]"
            type="button"
            variant="light"
          >
            admin
          
          </b-button-stub>
        </dd>
      </dl>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <eventformcontacts-stub
        eventid="1"
      />
    </b-col-stub>
  </b-row-stub>
</div>
`;
